src/test/java/org/cactoos/proc/ForEachTest.java

```diff
@@ -35,28 +36,23 @@
  * Test case for {@link ForEach}.
  *
  * @since 1.0
- * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
  */
-public class ForEachTest {
+@SuppressWarnings("unchecked")
+final class ForEachTest {
 
     @Test
-    void testProcIterable() throws Exception {
-        final List<Integer> list = new LinkedList<>();
-        new ForEach<Integer>(
-            list::add
-        ).exec(
-            new IterableOf<>(
-                1, 1
-            )
+    void worksWithGenerics() throws Exception {
+        final Collection<? super Iterable<?>> list = new LinkedList<>();
+        final Proc<? super Collection<? extends Number>> proc = list::add;
+        final List<List<Integer>> input = new ListOf<>(
+            new ListOf<>(1), new ListOf<>(2), new ListOf<>(3)
         );
+        new ForEach<>(proc).exec(input);
```

Comment:
@andreoss sorry to be nitpicky, but I think the point of using ? super or ? extends in the declaration of the ForEach class, is so that we don't have to do it here. The objective of this test should be to illustrate what a real user could write effortlessly. Nobody wants to bother with ? super or ? extends everywhere here :)

I propose the following instead, the type of `list` does not really matter, even List<?> is ok, but for the sake of the example, I put something useful for the assertion after:
```java
List<Iterable<Number>> list = new ListOf<>();
Proc<Iterable<Number>> proc = list::add;
new ForEach<Collection<Integer>>(proc).exec(new ListOf<>(1, 2, 3));
```